### ISSUE TITLE
Import detection fails if package path does not match package name

### ISSUE THREAD

**Issue by arekkas** [go-swagger/go-swagger#1587]
## Problem statement

When importing a package, such as `"gopkg.in/square/go-jose.v2"` (which exports `jose`) and using it in, for example, a `swagger:model` definition, then `swagger generate spec -m` throws a `no import found for jose`.

```go
import (
	"gopkg.in/square/go-jose.v2"
)

// Client represents an OAuth 2.0 Client.
//
// swagger:model oAuth2Client
type Client struct {
	JSONWebKeys *jose.JSONWebKeySet `json:"jwks,omitempty"`
}
```

A workaround for this is to name the import:

```
import (
	jose "gopkg.in/square/go-jose.v2"
)
```

But obviously, this should be properly resolved in the AST scan. Another thing is that this is quite difficult to debug as there is zero context (no filename) so it's really hard to find the location that caused the error. This could be improved by returning some context in `schema.go` with, for example:

```
		// find actual struct
		if selPath == "" {
			return nil, fmt.Errorf("no import found for %s in file %s.go", pth.Name, gofile.Name)
		}
```

## Environment
swagger version: current master
go version: 1.10

**Comment by fredbi** [comment]
It's a name mangling issue.
Packages are assumed to only exist out 1 word so they want the package to be like `github.com/sul-dslss-labs/permissions/service`

See also #1432

**Comment by arekkas** [comment]
The package name is one word (`jose`), the import path however is `go-jose`. This is not the only popular package in the Go ecosystem that does that, and `go-swagger` should be able to properly identify those packages as this type of naming is supported by Go.

**Comment by casualjim** [comment]
that's true but we just lean on goimports, I suggest this issue gets reported on that repo

**Comment by Radon22** [comment]
I faced with this issue too, I use ```import "github.com/paulmach/go.geojson"``` but package name is "geojson"
I found workaround ```import geojson "github.com/paulmach/go.geojson"``` but it is not applicable for third-party libraries
